src/modules/consumer.js

```diff
@@ -93,119 +86,137 @@ class Consumer {
     // ex: service-something with suffix :ci becomes service-suffix:ci etc.
     const suffixedQueue = `${queue}${this._connection.config.consumerSuffix || ''}`;
 
-    return this._connection
-      .getChannel(queue, options.channel || {})
-      .then((channel) => {
-        // when channel is closed, we want to be sure we recreate the queue ASAP so we trigger a reconnect by recreating the consumer
-        channel.addListener('close', () => {
-          this.subscribe(queue, options, callback);
-        });
+    const channel = await this._initializeChannel(queue, options || {}, callback);
+    if (!channel) {
+      // in case of any error creating the channel, wait for some time and then try to reconnect again (to avoid overflow)
+      await utils.timeoutPromise(this._connection.config.timeout);
+      return this.subscribe(queue, options, callback);
+    }
 
-        return channel
-          .assertQueue(suffixedQueue, options)
-          .then((q) => {
-            this._connection.config.transport.debug(loggerAlias, 'init', q.queue);
-            this._connection.config.logger.debug({
-              message: `${loggerAlias} init ${q.queue}`,
-              params: { queue: q.queue },
-            });
-
-            return this._consumeQueue(channel, q.queue, callback);
-          })
-          .catch((error) => {
-            this._connection.config.transport.error(loggerAlias, error);
-            this._connection.config.logger.error({
-              message: `${loggerAlias} Failed to assert queue ${queue}: ${error.message}`,
-              error,
-              params: { queue },
-            });
-          });
-        // in case of any error creating the channel, wait for some time and then try to reconnect again (to avoid overflow)
-      })
-      .catch((err) => {
-        if (err instanceof ChannelAlreadyExistsError) {
-          throw err;
-        } else {
-          return utils
-            .timeoutPromise(this._connection.config.timeout)
-            .then(() => this.subscribe(queue, options, callback));
-        }
+    try {
+      await channel.assertQueue(suffixedQueue, options);
+    } catch (error) {
```

Comment:
Previously if assert queue failed a retry would happen (with timeout in between) but I don't see it's happening here now.

Reply:
If the `assertQueue` failed because the connection was closed, the retry would happen on the `close` event listener that is added on the `_initializeChannel` function.

The "wait and reconnect" pattern only happens if the initialization itself failed, either for the connection or the channel, before connection/channel opens.

The function has always returned `true` regardless of the success/retry/failure status.